### ISSUE TITLE
Script will fail for at least 10 hours

### ISSUE THREAD

**Issue by VAsHachiRoku** [MicrosoftDocs/ATADocs#525]
The script creates a new AD group, then adds computers to this group, then the next line is to install the gMSA accounts on the endpoint. This is going to fail and cause a lot of confusion for many people.

1.) The installation will always fail for at least 10 hours because the current Kerberos Ticket does not have the newly created group membership as part of the already issued Ticket. When running the cmdlet Install-AdServiceAccount you'll receive "Access Denied" unless you reboot the endpoint and or klist to kill all current tickets. Best to use gpresult -r to see if the computer is a member of the newly created group.

2.) The line to install is missing a foreach loop, because this needs to be installed on all the endpoints that were added to the newly created group. https://docs.microsoft.com/en-us/powershell/module/activedirectory/install-adserviceaccount "The Install-ADServiceAccount cmdlet installs an existing Active Directory managed service account on the computer on which the cmdlet is run" in the example above this would need to be executed on DC01-06, plus the 2 ADFS endpoints. It would be better to foreach loop with invoke-command because the Install-ADServiceAccount cmdlet doesnt offer a -computer parameter to run remotely.

3.) A better example would be to use the builtin "Domain Controllers" and "Read-Only Domain Controllers" AD groups and explicitly add the two ADFS servers instead of creating another AD group to manage. This will also ensure the installation of the account #2 doesnt have to wait for refresh of TGT and or a reboot right away.

New-ADServiceAccount -Name $gMSA_AccountName -DNSHostName "$gMSA_AccountName.$env:USERDNSDOMAIN" `
-PrincipalsAllowedToRetrieveManagedPassword "Domain Controllers","Read-Only Domain Controllers","ADFS01","ADFS02"

Right now, if anyone were to copy and paste that script into a lab for testing it would cause more confusion as it's going to fail for at least 10 hours and the "Access Denied" would stump most people who are not aware that the Kerberos Ticket had not refreshed before trying the installation.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 12a9a5e8-afc3-7554-6a0b-248fae64a53f
* Version Independent ID: 12a9a5e8-afc3-7554-6a0b-248fae64a53f
* Content: [Directory Service Account recommendations](https://docs.microsoft.com/en-us/defender-for-identity/directory-service-accounts)
* Content Source: [ATPDocs/directory-service-accounts.md](https://github.com/Microsoft/ATADocs/blob/main/ATPDocs/directory-service-accounts.md)
* Service: **microsoft-defender-for-identity**
* GitHub Login: @dcurwin
* Microsoft Alias: **dacurwin**

**Comment by dcurwin** [comment]
Thanks for your comment. We'll investigate and get back to you.

**Comment by dcurwin** [comment]
@VAsHachiRoku  - Thanks again for your comment. We've updated the documentation. 

The product team has informed me that we didn't mean to have a straight end-to-end copy-paste script for a lab environment, as we need customers to understand what they run on their environment.
BTW, We already addressed the error in testing the gMSA (in the note), so the script wouldn't really "fail for at least 10 hours"
